lib/battle/CBattleInfoCallback.cpp

```diff
@@ -1611,6 +1627,16 @@ bool CBattleInfoCallback::battleHasDistancePenalty(const IBonusBearer * shooter,
 	return true;
 }
 
+bool CBattleInfoCallback::isEnemyUnitWithinSpecifiedRange(BattleHex attackerPosition, const battle::Unit * defenderUnit, unsigned int range) const
+{
+	for(auto hex : defenderUnit->getHexes())
+		if(BattleHex::getDistance(attackerPosition, hex) <= range)
+			return true;
+
+	//TODO what about shooting distance calculation for two-hex shooters? Is it correct?

```

Comment:
Checked H3 and yes, this is correct.
For double-wide shooters, game uses position of their "head", which matches what this code does